meta-refkit/recipes-image/images/refkit-installer-image.bb

```diff
@@ -191,29 +191,28 @@ REFKIT_INSTALLER_UEFI_COMBO () {
             fatal "sgdisk $output has failed - damaged disk?"
         fi
 
-        # Read partition description from rootfs.
-        if ! . "$input_mountpoint/boot/emmc-partitions-data"; then
-            fatal "reading $input_mountpoint/boot/emmc-partitions-data failed"
-        fi
+        PART_COUNT=$(execute sgdisk -p "$input" | tail -1 | awk '{print $1}')
 
         # Create partitions.
-        pnum=0
         gdisk_pnum=1
-        while [ "$pnum" -lt "$PART_COUNT" ]; do
-            eval size="\$PART_${pnum}_SIZE"
-            eval uuid="\$PART_${pnum}_UUID"
-            eval type_id="\$PART_${pnum}_TYPE"
-            eval lname="\$PART_${pnum}_NAME"
-            eval fs="\$PART_${pnum}_FS"
+        while [ "$gdisk_pnum" -le "$PART_COUNT" ]; do
+            size=$(execute sgdisk -i $gdisk_pnum "$input" | grep ^"Partition size" | cut -d : -f 2 | awk '{print $1}')
+            type_id=$(execute sgdisk -i $gdisk_pnum "$input" | grep ^"Partition GUID code" | cut -d : -f 2 | awk '{print $1}')
+            lname=$(execute sgdisk -i $gdisk_pnum "$input" | grep ^"Partition name" | cut -d : -f 2 | awk '{print $1}' | tr -d \')
 
+            if [ "$lname" = "rootfs" ]; then
+                uuid=$(execute grep PARTUUID "$input_mountpoint"/boot/EFI_internal_storage/BOOT/boot*.efi | sed -e 's/.*PARTUUID=\([a-zA-Z0-9-]*\).*/\1/')
```

Comment:
This is non-obvious and should come with a comment about REMOVABLE_MEDIA_ROOTFS_PARTUUID_VALUE vs. INT_STORAGE_ROOTFS_PARTUUID_VALUE  and how we ensure here that we use the value expected by the kernel.

Also, instead of plain grep it might be better to use "strings "$input_mountpoint"/boot/EFI_internal_storage/BOOT/boot*.efi | grep PARTUUID". IMHO strings is more likely to output one string per line than grep.